app/src/main/java/com/github/gotify/messages/MessagesActivity.java

```diff
@@ -178,6 +180,11 @@ protected void onUpdateApps(List<Application> applications) {
 
     private Picasso makePicasso() {
         OkHttpClient.Builder builder = new OkHttpClient.Builder();
+        int CACHE_SIZE = 10 * 1024 * 1024; // 10 MB
```

Comment:
The current solution will work, if the webserver sends the cache-control header with max-age. Do you think we can rely on this or should we added some kind of default cache timeout for images that are served without cache-control?

Reply:
The main images to get updated are the Application images... and those images don't get "updated" -- they are instead replaced with new images with new filenames. This means that any call that refreshes the application list will also get any new application icon image references and the original cached version will just get ignored.

In practice, reloading the app or even just switching between portrait/landscape will cause MessagesActivity.onCreate to get called, refreshing the application list and any updated Application images will get loaded.

The only images that won't get updated via an application list refresh are images that are embedded in messages with Markdown. The App-drawer refresh button will cause these to get downloaded though.

Reply:
Just realized the images could get updated directly on the server outside of the API -- i.e. updated without a filename change. Restarting the app wouldn't refresh those changes on the client, but the app-drawer onRefreshAll call will. Seems like a good compromise.